src/lib/receiver.js

```diff
@@ -1,15 +1,17 @@
 'use strict'
 
-const crypto = require('crypto')
 const uuid = require('uuid')
 const moment = require('moment')
 const Client = require('ilp-core').Client
 const EventEmitter = require('eventemitter2')
+const base64url = require('../utils/base64url')
+const cc = require('../utils/condition')
 const debug = require('debug')('ilp:receiver')
 const BigNumber = require('bignumber.js')
 const createHmacHelper = require('../utils/hmac').createHmacHelper
 
-const RECEIVER_ID_PREFIX = '~ipr.'
+const IPR_RECEIVER_ID_PREFIX = '~ipr.'
+const SSP_RECEIVER_ID_PREFIX = '~ssp.'
```

Comment:
As per another comment - they both hash the "request"/"packet" the same way - so we could split them into separate modules, but there would be a lot of overlap. You can kind of tell from this PR - there are only a handful of lines actually dedicated to only IPR or only SSP, 95% of the sender and receiver code is the same for both.

So we should chat about it - but it seems like having two different receivers seems like we're overengineering it a bit.

> Chances are you won't be using both at the same time and I think it could cause issues having the same receiver instance listening on two separate protocols.

I don't think so - at least if we make it so that SSP requires a callback to be registered with the receiver. If you only want to use IPR, you're fine, because there is no callback registered, so this receiver will reject SSP payments. And if you only want to use SSP, you're fine because you haven't issued any payment requests, so you won't get any valid IPR payments.

(The latter is a bit sketchier, because it relies on cryptography. Personally, I think it would be fine. But if we think it isn't, we could just add a method to enable IPR.)